browser/src/react-redux/App/Login/Login.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+
+export default class Login extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isValidated: false,
+      validEmail: false,
+      validPassword: false,
+      email: '',
+      password: '',
+      emailError: '',
+      passwordError: ''
+    }
+    this.handleInputChange = this.handleInputChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.displayErrors = this.displayErrors.bind(this)
+  }
+
+  handleInputChange(event) {
+    const target = event.target
+    const value = target.value
+    const name = target.name
+
+    this.setState({
+      [name]: value
+    })
+
+    //validation checks
+    if (name === 'email') {
+      this.validateEmail(value)
+        ? this.setState({ validEmail: true })
+        : this.setState({ validEmail: false })
+    } else {
+      this.validatePassword(value)
+        ? this.setState({ validPassword: true })
+        : this.setState({ validPassword: false })
+    }
+    //check if both email and password are valid
+    this.isValidatedCheck()
+  }
+
+  validateEmail(value) {
+    //just checks for an @ for now
+    const atCheck = /\S+@\S+/
+    return value.length >= 3 && atCheck.test(value) ? true : false
+  }
+  validatePassword(value) {
+    //just check length for now
+    return value.length >= 6 ? true : false
+  }
+
+  isValidatedCheck() {
+    this.state.validEmail && this.state.validPassword
+      ? this.setState({ isValidated: true })
+      : this.setState({ isValidated: false })
+  }
+
+  displayErrors() {
+    //displayErrors() will update the state for the corresponding input that has not satisfied the validation reqs.
+    if (!this.state.validEmail && !this.state.validPassword) {
+      this.setState({ emailError: 'Please enter a valid e-mail:' })
+      this.setState({ passwordError: 'Please enter a valid password:' })
+    } else if (!this.state.validPassword && this.state.validEmail) {
+      this.setState({ passwordError: 'Please enter a valid password:' })
+      this.setState({ emailError: '' })
+    } else if (!this.state.validEmail && this.state.validPassword) {
+      this.setState({ emailError: 'Please enter a valid e-mail:' })
+      this.setState({ passwordError: '' })
+    } else {
+      this.setState({ emailError: '', passwordError: '' })
+    }
+  }
+
+  handleSubmit(e) {
```

Comment:
We want to make an api call here. Go ahead and use `axios` for this. It's already been saved to your `package.json` file.

If the response is successful, we will want to:
1. update the user in the redux store by calling an action (hold off on this until the signup form is merged in so you can use it as a reference)
2. we will also want to redirect to the hompage `/`

If the response returns an error, we will want to display those errors above or below the input field. I just added comments to #24 to give you examples of what the success/failure responses structures will look like.